tools/swift-format/Sources/Rules/CommaWhitespace.swift

```diff
@@ -14,5 +14,40 @@ import SwiftSyntax
 ///
 /// - SeeAlso: https://google.github.io/swift#horizontal-whitespace
 public final class CommaWhitespace: SyntaxFormatRule {
+  public override func visit(_ token: TokenSyntax) -> Syntax {
+    guard let next = token.nextToken else { return token }
+    
+    /// Commas own their trailing spaces, so ensure it only has 1 if there's
+    /// another token on the same line.
+    if token.tokenKind == .comma,
```

Comment:
When dealing with Boolean conditions, use `&&` to join them instead of a comma. The comma is meant to be used when joining `if let` clauses.

So this would be `if token.tokenKind == .comma && !next.leadingTrivia.containsNewlines`.